Tests/LegacyUnit/Service/RegistrationManagerTest.php

```diff
@@ -2212,6 +2225,39 @@ public function notifyAttendeeForSendConfirmationFalseNotCallsModifyThankYouEmai
         $this->subject->notifyAttendee($registration, $pi1);
     }
 
+    /**
+     * @test
+     */
+    public function notifyAttendeeForSendConfirmationTrueCallsRegistrationEmailHookMethodsForPlainTextEmail()
+    {
+        \Tx_Oelib_ConfigurationProxy::getInstance('seminars')
+            ->setAsInteger('eMailFormatForAttendees', TestingRegistrationManager::SEND_TEXT_MAIL);
+
+        $registration = $this->createRegistration();
+
+        $hook = $this->createMock(RegistrationEmail::class);
+        $hook->expects(self::once())->method('modifyAttendeeEmail')
+            ->with(self::anything(), self::anything(), 'confirmation');
+        $hook->expects(self::once())->method('modifyAttendeeEmailBodyPlainText')
+            ->with(self::anything(), $registration, 'confirmation');
+        $hook->expects(self::never())->method('modifyAttendeeEmailBodyHtml');
+        $hook->expects(self::never())->method('modifyOrganizerEmail');
+        $hook->expects(self::never())->method('modifyAdditionalEmail');
+
+        $hookClass = \get_class($hook);
+        $GLOBALS['TYPO3_CONF_VARS']['EXTCONF']['seminars'][RegistrationEmail::class][] = $hookClass;
+        GeneralUtility::addInstance($hookClass, $hook);
+
+        $this->subject->setConfigurationValue('sendConfirmation', true);
+        $pi1 = new \Tx_Seminars_FrontEnd_DefaultController();
+        $pi1->init();
+
+        $this->subject->notifyAttendee($registration, $pi1);
+
+        // purge added instance in case of failing test
+        $this->mockedClassName = $hookClass;
```

Comment:
For all tests, please let's move this directly to below the `addInstance` call so any exceptions etc. in the code do not break the cleanup.

Reply:
I refactored that to be more generic and as a direct drop-in replacemant for `GeneralUtility::makeInstance()`: `$this->addMockedInstance()`.